lambdas/questions/src/test/java/uk/gov/di/ipv/cri/experian/kbv/api/QuestionHandlerTest.java

```diff
@@ -0,0 +1,113 @@
+package uk.gov.di.ipv.cri.experian.kbv.api;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.amazonaws.xray.AWSXRay;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.di.ipv.cri.experian.kbv.api.domain.PersonIdentity;
+import uk.gov.di.ipv.cri.experian.kbv.api.domain.QuestionRequest;
+import uk.gov.di.ipv.cri.experian.kbv.api.domain.QuestionsResponse;
+import uk.gov.di.ipv.cri.experian.kbv.api.domain.ValidationResult;
+import uk.gov.di.ipv.cri.experian.kbv.api.service.KBVService;
+import uk.gov.di.ipv.cri.experian.kbv.api.service.KBVServiceFactory;
+import uk.gov.di.ipv.cri.experian.kbv.api.service.KeyStoreService;
+import uk.gov.di.ipv.cri.experian.kbv.api.validation.InputValidationExecutor;
+
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class QuestionHandlerTest {
+    private QuestionHandler questionHandler;
+    @Mock private KBVServiceFactory kbvServiceFactoryMock;
+    @Mock private KBVService kbvServiceMock;
+    @Mock private ObjectMapper objectMapperMock;
+    @Mock private KeyStoreService keyStoreServiceMock;
+    @Mock private InputValidationExecutor inputValidationExecutorMock;
+
+    @BeforeEach
+    void setUp() {
+        AWSXRay.beginSegment("handleRequest");
```

Comment:
We are not interested in testing / executing any of the AWSXRay code therefore this line can be removed and you can add the following to the class default constructor or add a static initialiser:
`// prevent aws-xray missing context exception from surfacing during unit test execution`
`System.setProperty("com.amazonaws.xray.strategy.contextMissingStrategy", "LOG_ERROR");`

Reply:
Thanks I avoiding the static initializer mainly for testing reasons, I think 

```
        AWSXRay.setGlobalRecorder(AWSXRayRecorderBuilder.standard().withContextMissingStrategy(new LogErrorContextMissingStrategy()).build());

```
This is equivalent to the above